### DISCUSSION TITLE
Can potions become an equip-able item (like Path of Exile)?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#864] [Q&A] [answered]
Can potions become an equip-able item (like Path of Exile) in the current Flare engine?

If the Flare engine can have the option for potions to become an equip-able item (like Path of Exile or kinda with Baldur's Gate) then that could limit potion spam where say only 3 to 6 (a limited amount of potion slots) can be drank at the same time with cool downs per each potion type there could be a limited burst of drinking one of each potion type equipped before waiting for cool downs. The equipped potion types could and mixed and matched on the 'potion belt' / potion equip-able areas / 3-6 ~potion slots. Also say a stack of ~10 of each kind could be in order so drinking many potions can still happen but also to keep things reasonable and avoid infinite potions / free resources.

Can potions be made only usable if equipped and still consumable? Like how they currently work but can only be activated / 'drank' if they are equipped first?

**Comment by dorkster** [comment, marked as answer]
Interesting idea. I tried it out and it "just works" for the most part. Here's what I did:

- Added `equipment_slot=256,160,potion` to `menus/inventory.txt`
- Added `item_type=potion` to the Health Potion item (id=2)
- Changed `requires_item=2` to `requires_equipped_item=2,1` in the potion's power definition (`powers/base/items/health_potion.txt`)

Observations:

- The quantity of potions isn't shown on the action bar. We assumed equipped items would always have a quantity of 1, but this should be easily fixed.
- I think it would be nice for quality-of-life if the stack of equipped potions was automatically replenished from the player's inventory. Otherwise, the player has to re-equip the potion when they run out.

**Reply by WithinAmnesia** [reply]
That is great sounding progress @dorkster 

With your set up can you confirm that potions only can be used / 'drank' if they are equipped and they cannot be used / 'drank' from the inventory? I hope that the potion use / 'drink' ability is restricted enough so that players cannot drink potentially hundred of different potion types from the inventory during a fight and potentially ruin the combat experience by being almost invincible from effectively avoiding cooldowns.

I too think it would be nice for quality-of-life if the stack of equipped potions was automatically replenished from the player's inventory. Otherwise, the player has to re-equip the potion when they run out.

**Reply by dorkster** [reply]
Yes, it works correctly. Trying to "use" a potion by right clicking on it in the inventory will just equip it. And the power is correctly disabled when the potion equip slot is empty, even if the player is carrying more potions.

**Reply by WithinAmnesia** [reply]
Excellent! I will have to run some tests :-)!
